### PR TITLE
Fix test warnings; remove cruft

### PR DESCRIPTION
**Pull request by jdavies-st**
Fixes the following warning
```
./site-packages/_pytest/mark/structures.py:332
  ./site-packages/_pytest/mark/structures.py:332: PytestUnknownMarkWarning: Unknown pytest.mark.importorskip - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,
```
Plus some updates to testing infrastructure.

### CODE REVIEW

**Comment by nden** [comment]
@jdavies-st Can you rebase, it looks like some duplicate changes are included. And may be squash?

**Comment by jdavies-st** [comment]
Successful rebase. The 2 schema failures show up now with the released version of `astropy`, but pass with `astropy/master`.  Would be interested in feedback of adding `gwcs` to the test deps, i.e. the `[test]` install extras.

File: conftest.py [hunk 1]
```diff
@@ -1,35 +1,10 @@
...
-    display.PYTEST_HEADER_MODULES['gwcs'] = 'gwcs'
-except ImportError:
-    pass
-
-
-pytest_plugins = [
-    'astropy.tests.plugins.display'
-]
-
-enable_deprecations_as_exceptions()
```
Comment (nden) [inline conftest.py]: Is this defined somewhere else or is it a change in testing?

Comment (jdavies-st) [reply conftest.py]: This removes this functionality and dependency on `astropy.tests.helpers` and `astropy.tests.plugins.display`.

That said `enable_deprecations_as_exceptions()` doesn't seem to work.  I.e. we are seeing deprecations yet the tests are passing.  So I'm not sure how useful this is.  The better way to do this is
```
pytest -W error::DeprecationWarning
```
which correctly catches deprecations exceptions.

**Pull request merged**